Kerberos.NET/Asn1/Experimental/CryptoPool.cs

```diff
@@ -26,31 +27,39 @@ internal static void Return<T>(T[] array, int clearSize = ClearAll)
 
             if (!clearWholeArray && clearSize != 0)
             {
-                Array.Clear(array, 0, clearSize);
+                ZeroMemory(array.AsSpan(0, clearSize));
             }
 
             ArrayPool<T>.Shared.Return(array, clearWholeArray);
         }
+
+        [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
+        private static void ZeroMemory<T>(Span<T> buffer)
+        {
+            // NoOptimize to prevent the optimizer from deciding this call is unnecessary
+            // NoInlining to prevent the inliner from forgetting that the method was no-optimize
+            buffer.Clear();
+        }
     }
 
     internal struct CryptoMemoryOwner<T> : IMemoryOwner<T>
     {
-        private readonly T[] memory;
-        private readonly bool clearAll;
+        private readonly T[] _memory;
```

Comment:
Why the rename? It's fine, just curious?

Reply:
`this.clearAll = clearAll` from old L44 led me to this, so the same name for the field and the parameter.
Also coding-rules in _dotnet_ (I know PRs shouldn't change it, as it add noise).